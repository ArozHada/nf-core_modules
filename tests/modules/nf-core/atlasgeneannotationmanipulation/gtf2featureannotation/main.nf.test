nextflow_process {

    name "Test Process ATLASGENEANNOTATIONMANIPULATION_GTF2FEATUREANNOTATION"
    script "modules/nf-core/atlasgeneannotationmanipulation/gtf2featureannotation/main.nf"
    process "ATLASGENEANNOTATIONMANIPULATION_GTF2FEATUREANNOTATION"
    config "tests/modules/nf-core/atlasgeneannotationmanipulation/gtf2featureannotation/nextflow.config"
    tag "atlasgeneannotationmanipulation_gtf2featureannotation"

    test("Human Genome GTF") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ "id":"homo_sapiens"],
                            file(params.test_data['homo_sapiens']['genome']['genome_gtf'], checkIfExists: true) 
                           ]
                input[1] = [
                            [ "id":"homo_sapiens"],
                            []
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("human_genome_gtf") }
            )
        }
    }

    test("Human Genome GTF With Fasta") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ "id":"homo_sapiens"],
                            file(params.test_data['homo_sapiens']['genome']['genome_gtf'], checkIfExists: true) 
                           ]
                input[1] = [
                            [ "id":"homo_sapiens"],
                            file(params.test_data['homo_sapiens']['genome']['transcriptome_fasta'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("human_genome_gtf_fasta") }
            )
        }
    }
}
