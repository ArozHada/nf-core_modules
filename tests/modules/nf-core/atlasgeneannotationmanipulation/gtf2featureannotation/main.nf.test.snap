{
    "human_genome_gtf_fasta": {
        "content": [
            [
                "homo_sapiens.anno.tsv:md5,f064258c7741d1ca0c9ff3492797a835",
                "versions.yml:md5,a6f9622615b98b2c777db8c7d56b3ff8"
            ]
        ],
        "timestamp": "2023-04-07T21:57:31+0000"
    },
    "human_genome_gtf": {
        "content": [
            [
                "homo_sapiens.anno.tsv:md5,f064258c7741d1ca0c9ff3492797a835",
                "versions.yml:md5,a6f9622615b98b2c777db8c7d56b3ff8"
            ]
        ],
        "timestamp": "2023-04-07T21:57:31+0000"
    }
}